db.go

```diff
@@ -0,0 +1,73 @@
+package auth
+
+import (
+	"errors"
+
+	"github.com/jinzhu/gorm"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+)
+
+const DefaultMaxIdleConns = 5
+
+var ErrorNoConnectionString = errors.New("A connection string must be specified on the first call to Get")
+
+// DatabaseHandler abstracts away common persistence operations needed for this package
+type DatabaseHandler interface {
+	// GetUser gets a user from the database that matches constraints on the input user
+	GetUser(u User) (User, error)
+	// UpsertUser updates a user (if input user UUID matches one in the db) or inserts a user
+	UpsertUser(u User) error
+}
+
+type databaseHandler struct {
+	db *gorm.DB
+}
+
+// NewDatabaseHandler initializes and returns a new DatabaseHandler
+func NewDatabaseHandler(dbConnection string) (DatabaseHandler, error) {
+	db, err := getDB(dbConnection)
+	if err != nil {
+		return nil, err
+	}
+	// AutoMigrate relevant schemas
+	db.AutoMigrate(&User{})
+
+	return &databaseHandler{
+		db: db,
+	}, nil
+}
+
+func (a *databaseHandler) GetUser(u User) (User, error) {
+	var foundUser User
+	err := a.db.Where(&u).First(&foundUser).Error
+	if err != nil {
+		return User{}, err
+	}
+
```

Comment:
How about this syntax
```go
if err := a.db.Where(&u).First(&foundUser).Error; err != nil {
    return User{}, err
}
```
literally the favorite thing i found

Reply:
haha I do like this sometimes when the line length doesn't get too long...

Reply:
On [L50](https://github.com/suyashkumar/auth/pull/7/files#diff-7a117cbfd5485ed35efcbb2995c37aa0R51) this pattern would get too long, and would rather keep it consistent between here and there so will leave as is (I think it's more readable too) :).